doc/dynamic-environments/configuration.mkd

```diff
@@ -237,10 +237,16 @@ repo would no longer cause the matching environment to be purged.
 
 When a given environment is deployed, recursively remove any content found
 which is not either committed to the control repo branch that maps to that
-environment, or declared in a Puppetfile committed to that branch. Note that
-the .r10k-deploy.json file is exempt from this purging. Additionally, no
-content will be purged if an error is encountered while deploying the
-environment's Puppetfile content.
+environment, or declared in a Puppetfile committed to that branch.
```

Comment:
I think this "not... either... or..." is exactly why "neither... nor..." exists.